shopping-cart/src/pages/CartPage/CartPage.jsx

```diff
@@ -0,0 +1,280 @@
+import React, { useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import API from '../../API';
+import './CartPage.sass'
+
+// #### 장바구니
+// ​
+// - [x] 해당 상품의 수량을 변경할 수 있다.
+//   - [x] 상품의 수량은 항상 1이상, 20이하여야 한다
+//     - [x] 상품의 수량이 1이면 상품 수량 감소할 수 없다.
+//     - [x] 상품의 수량이 20이면 상품 수량 증가할 수 없다.
+//   - [x] 해당 상품의 총 금액이 변경된다.
+//   - [x] 해당 상품이 체크되어있으면, 결제예상금액도 변경된다.
+// - [x] 모두선택 버튼이 체크되면, 상품들이 모두 선택된다.
+//   - [x] 모두선택 버튼이 체크가 풀리면, 상품들의 선택이 모두 해제된다.
+// - [x] 상품 삭제 버튼을 누르면, confirm 메시지가 보여진다.
+//   - [x] 확인을 누르면, 선택된 상품이 모두 삭제된다.
+//   - [x] 결제예상금액이 0원이 된다.
+// - [x] �� 버튼을 누르면 confirm 메시지가 보여진다.
+//   - [x] 확인을 누르면, 해당 상품이 삭제된다.
+// - [x] 체크된 상품 개수에 따라 주문하기 버튼 내부에 수량이 변경된다.
+// - [x] 주문하기 버튼을 누르면, confirm 메시지가 보여진다.
+//   - [x] 확인을 누르면, 주문/결제 페이지로 이동한다.
+//   - [x] 확인을 누르면, 장바구니에서 선택된 상품들이 삭제된다.
+//   - [x] 확인을 누르면, 체크된 상품들을 데이터베이스에서 제거한다.
+// - [x] 주문할 상품이 0개이면 버튼이 비활성화된다.
+/*
+const productInfo = {
+  id,
+  product: {},
+  quantity,
+  checked
+  ==>
+  id,
+  price,
+  name,
+  imageUrl,
+  quantity
+}
+*/
+const CartPage = () => {
+  const [cart, setCart] = useState([]);
+  const navigate = useNavigate();
+
+  const updateCart = async() => {
+    await API.getProduct("/carts")
+      .then(res => res.data)
+      .then(datas => datas.map(data => {
+        const {id, product} = data;
+        const information = {
+          id,
+          price: product.price,
+          name: product.name,
+          imageUrl: product.imageUrl,
+          quantity: 1,
+          checked: true
+        }
```

Comment:
```suggestion
        const {id, product} = data;
        const { price, name, imageUrl} = product;
        
        const information = {
          id,
          price,
          name,
          imageUrl,
          quantity: 1,
          checked: true
        }
```

이렇게 하면 더 잘 의미 전달이 되지 않을까 싶습니다!

Reply:
이후에 이렇게 수정했습니다.
처음에 급하게 구현하려다보니...그렇게 된 것같습니다.
말씀해주신 방법으로 나타낸게 나중에 data넘길때도 편리했던 것 같습니다.